atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/TransientRowsWorkflows.java

```diff
@@ -0,0 +1,181 @@
+import com.palantir.atlasdb.workload.invariant.Invariant;
+import com.palantir.atlasdb.workload.invariant.InvariantReporter;
+import com.palantir.atlasdb.workload.store.ImmutableWorkloadCell;
+import com.palantir.atlasdb.workload.store.InteractiveTransactionStore;
+import com.palantir.atlasdb.workload.store.ReadableTransactionStore;
+import com.palantir.atlasdb.workload.store.TableAndWorkloadCell;
+import com.palantir.atlasdb.workload.store.WorkloadCell;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedReadTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransaction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransactionAction;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public final class TransientRowsWorkflows {
+    private static final SafeLogger log = SafeLoggerFactory.get(TransientRowsWorkflows.class);
+
+    @VisibleForTesting
+    static final int SUMMARY_ROW = -1;
+
+    @VisibleForTesting
+    static final int COLUMN = 1;
+
+    @VisibleForTesting
+    static final int VALUE = 0;
+
+    private static final SecureRandom SECURE_RANDOM = DefaultNativeSamplingSecureRandomFactory.INSTANCE.create();
+
+    private TransientRowsWorkflows() {}
+
+    public static Workflow create(
+            InteractiveTransactionStore store,
+            TransientRowsWorkflowConfiguration workflowConfiguration,
+            ListeningExecutorService executionExecutor) {
+        return DefaultWorkflow.create(
+                store,
+                (txn, taskIndex) -> run(store, workflowConfiguration, taskIndex),
+                workflowConfiguration,
+                executionExecutor);
+    }
+
+    private static Optional<WitnessedTransaction> run(
+            InteractiveTransactionStore store, TransientRowsWorkflowConfiguration configuration, int taskIndex) {
+        return store.readWrite(txn -> {
+            String tableName = configuration.tableConfiguration().tableName();
+            txn.write(tableName, ImmutableWorkloadCell.of(SUMMARY_ROW, taskIndex), VALUE);
+            txn.write(tableName, ImmutableWorkloadCell.of(taskIndex, COLUMN), VALUE);
+            if (taskIndex > 0) {
+                int indexToRead = SECURE_RANDOM.nextInt(taskIndex);
+                txn.read(tableName, ImmutableWorkloadCell.of(SUMMARY_ROW, indexToRead));
+                txn.read(tableName, ImmutableWorkloadCell.of(indexToRead, COLUMN));
+
+                txn.delete(tableName, ImmutableWorkloadCell.of(SUMMARY_ROW, taskIndex - 1));
```

Comment:
We probably should check if the value exists _before_ issuing a delete; in production, I'd expect us to almost always to only delete rows that we know exist!